c/common/src/codingstandards/c/Pointers.qll

```diff
@@ -62,12 +62,7 @@ class ArrayPointerArithmeticExpr extends PointerArithmeticExpr, ArrayExpr {
 predicate isNullPointerConstant(Expr e) {
   e.findRootCause() instanceof NULLMacro
   or
-  exists(CStyleCast c |
-    not c.isImplicit() and
-    c.getExpr() = e and
-    e instanceof Zero and
-    c.getType() instanceof VoidPointerType
-  )
+  e instanceof Zero
```

Comment:
Note to self: this affects Rule 11.9, which is incorrect. `0` is a null pointer constant, but not permitted according to 11.9.

Reply:
I fixed the incompatibility with Rule 11.9, which didn't change the behaviour of Rule 11.9 itself. However, I did notice and fix a small issue, for which I have included a change note entry.